libmachine/util/util.go

```diff
@@ -0,0 +1,29 @@
+package util
+
+import (
+	"os"
+	"strings"
+
+	"github.com/rancher/machine/libmachine/log"
+)
+
+func FindEnvAny(names ...string) string {
+	for _, n := range names {
+		if val := os.Getenv(n); val != "" {
+			return val
+		}
+	}
+	return ""
+}
+
+func GetProxyURL() string {
+	urlRaw := FindEnvAny("HTTP_PROXY", "HTTPS_PROXY")
+	if urlRaw == "" {
+		log.Debug("env var HTTP_PROXY or HTTPS_PROXY is not found")
+		return ""
+	}
+	urlRaw = strings.ToLower(urlRaw)
+	urlRaw = strings.TrimPrefix(urlRaw, "http://")
+	urlRaw = strings.TrimPrefix(urlRaw, "https://")
```

Comment:
This can be done in one line with `regexp`. Not sure if it's worth it. Thoughts?

Reply:
It is implemented this way because 
- we are not sure if there are `http://` or `https://`  at the beginning of the url 
- we are not validating the format of the url ( do we need to?)
- it might be more "expensive" to use regexp than `strings.TrimPrefix`